meson.build

```diff
@@ -86,6 +87,19 @@ endif
 if libidn2_dep.found() or libidn_dep.found()
   # Check for libunistring, we need it for psl_str_to_utf8lower()
   libunistring = cc.find_library('unistring')
+  found_iconv = false
+  if cc.has_function('iconv_open')
+    libiconv_dep = []
+    found_iconv = true
+  endif
+  if not found_iconv and cc.has_header_symbol('iconv.h', 'iconv_open')
```

Comment:
Note that once you are able to upgrade your minimum meson requirement to 0.60, you can replace these 13 lines with just `dependency('iconv')`.